app/components/stepper.js

```diff
@@ -21,17 +25,33 @@ export default class StepperComponent extends Component {
   setIsValid = (newVal) => (this.isValid = newVal);
   setIsPreValid = (newVal) => (this.preValid = newVal);
 
+  constructor() {
+    super(...arguments);
+    if (
+      localStorage.getItem('currentStep') <=
+      new URLSearchParams(window.location.search).get('step')
+    ) {
+      this.router.transitionTo('join', {
+        queryParams: {
+          step: localStorage.getItem('currentStep'),
+        },
+      });
+    }
+  }
+
   @action incrementStep() {
     if (this.currentStep < 5) {
       this.currentStep += 1;
       localStorage.setItem('currentStep', this.currentStep);
+      window.history.pushState({}, '', `?step=${this.currentStep}`);
```

Comment:
Check the below snippet if it's good to go with as your using **Router** service
```jsx
    this.router.transitionTo({ queryParams: { paramName: 'paramValue' } });
 ```

Reply:
done